### PR TITLE
Georgi Shopov HW05

### PR DESCRIPTION
**Pull request by georgiiishopovv**

### CODE REVIEW

File: monads/src/HW/Eval.hs [hunk 1]
```diff
@@ -22,15 +22,51 @@ data MachineState v = MachineState
   }
   deriving (Show)
 
+instance (Eq v) => Eq (MachineState v) where
+   (MachineState s1 e1) == (MachineState s2 e2) = s1 == s2 && e1 == e2
```
Comment (kajigor) [inline monads/src/HW/Eval.hs]: Derive.

File: monads/src/HW/Eval.hs [hunk 2]
```diff
@@ -22,15 +22,51 @@ data MachineState v = MachineState
...
 initialState :: MachineState String
 initialState = MachineState [] M.empty
 
 -- Execute a single instruction. 
 -- Successful evaluation does not produce any useful result: only the effect of modifying state matters. 
 execInstr :: (Ord v, Show v) => StackInstr v -> MyState (MachineState v) (Either (Error v) ())
-execInstr = undefined 
+execInstr (PushNum n) = do
+   modify $ \state -> state {getStack = n : getStack state}
+   return (Right ())
```
Comment (kajigor) [inline monads/src/HW/Eval.hs]: You don't need this return.

File: monads/src/HW/Eval.hs [hunk 3]
```diff
@@ -22,15 +22,51 @@ data MachineState v = MachineState
...
+execInstr (PushVar v) = do
+   state <- get
```
Comment (kajigor) [inline monads/src/HW/Eval.hs]: Use `gets`

**Review by kajigor — CHANGES_REQUESTED** [review]

**Comment by kajigor** [comment]
Resolve the conflict

**Comment by georgiiishopovv** [comment]
> Resolve the conflict

Fixed <2 days ago>

**Comment by kajigor** [comment]
Ok

**Pull request closed without merging**